### PR TITLE
feat: add Tag component for Form Card

### PR DESCRIPTION
**Pull request by aalencar**
## Proposed API

```typescript
/** Create a set of tags laid out horizontally. */
export interface Tags {
  /** Tags in this set. */
  items: Tag[]
}
```

## Design

Large
image

Small
image

## Implementation

Default
image

H2O dark
image

Neon
image

### CODE REVIEW

**Comment by aalencar** [comment]
@mturoci Although the indentation for `info_tag.md` looks right, I couldn't check it running `tools/showcase make generate` because playwright is still failing for me.

Can we have all the QA checks (`ui make test`, `tools/showcase`, etc) in the CI?

**Review by mturoci — CHANGES_REQUESTED** [review]
Thanks @aalencar! A few comments from me.

>  I couldn't check it running tools/showcase make generate because playwright is still failing for me.

I updated Playwright to the latest version so try running

```sh
cd tools/showcase
make clean
make setup
```
If you still encounter some errors, feel free to reach out to me.

> Can we have all the QA checks (ui make test, tools/showcase, etc) in the CI?

`ui make test` could be added to PR checks, but not sure if worth the effort as it fails only when tinkering with state internals of a specific component in which case one has to use these tests during development anyway. For new features like this or small independent changes (which is most of the work) the tests will probably not fail anyway.

Moreover, we run it during nightly builds so we know even if something broke relatively fast.

For `tools/showcase`, there are 2 targets to be discussed 
* `make generate` (the one you need currently) - this is used for generating docs content and since you need to visually check it anyway, it would not be of any help to you within the CI
* `make visual-regression` this one is relatively flaky (could be resolved for the price of losing performance with explicit sleeps, but probably not worth it) and is used only when working on changes that basically affect whole codebase (deps bump, changes to how layouting works etc.), which is not so often. They also output A LOT of false positives so would be just a burden for general dev workflow if run for every simple PR. Also note that CI integration would take a lot of effort also since we would need to figure out the storage for snapshots (more than 2000) and also ensure they are generated on the very same machine as the screenshots are hardware (graphics) dependent.

**Comment by aalencar** [comment]
It was broken if the label had more than 1 character. Now it supports it.

image

File: ui/src/info_tag.tsx
Comment (aalencar) [reply ui/src/info_tag.tsx]: I don't see any performance improvement here but it makes sense to not recalculate the size every single render. Refactored. Thanks!

File: ui/src/info_tag.tsx
Comment (aalencar) [reply ui/src/info_tag.tsx]: Changed.

File: ui/src/info_tag.test.tsx
Comment (aalencar) [reply ui/src/info_tag.test.tsx]: I believe that visual regression is related to how it looks, making sure that changes to the logic don't impact the colors and other properties of our components. In this case, label is a requirement for our component and in this test I was checking if the value passed to the prop is properly rendered. I removed the test since we don't have it for other components.

File: website/components/form/info_tag.md
Comment (aalencar) [reply website/components/form/info_tag.md]: Done.

**Comment by aalencar** [comment]
Playwright is working for me now. Thanks @mturoci!

**Comment by aalencar** [comment]
Please, check if the changes to docs are good enough.

**Review by mturoci — DISMISSED** [review]
LGTM @aalencar.

Let's wait for API review from @lo5.

**Review by lo5 — CHANGES_REQUESTED** [review]
We already have a `ui.tag()` in the API (implementation is in `tag_table_cell_type.tsx`. 
I would recommend adding `size` to it and reusing it. The `name` attribute is not used in this PR's implementation anyway. ​

**Review by mturoci — CHANGES_REQUESTED** [review]
Thanks @aalencar, please update this PR with correct name for the component. Should be tag not info tag (source files/example/docs).

**Comment by aalencar** [comment]
It looks like there is a conflict of name to resolve `ui.tag` for Table and `ui.tag` for Form Card. Working on it.

**Comment by aalencar** [comment]
@lo5 any suggestions for the component's name? Should we stick with `info_tag`?

**Comment by lo5** [comment]
@aalencar Can't we reuse the existing `ui.tag()`? What's the need for a new component?

**Comment by aalencar** [comment]
@lo5 for the current `ui.tag` implemented for `Table` we have to define which tags are available when creating the column and then pass a comma-separated list of tags for each row.

```python
# Create some issues
issues = [Issue(text=fake.sentence(), tag=('FAIL' if i % 2 == 0 else 'DONE,SUCCESS')) for i in range(10)]

columns = [
    ui.table_column(name='text', label='Issue', min_width='400px'),
    ui.table_column(name='tag', label='Badge', cell_type=ui.tag_table_cell_type(name='tags', tags=[
        ui.tag(label='FAIL', color='$red'),
        ui.tag(label='DONE', color='#D2E3F8', label_color='#053975'),
        ui.tag(label='SUCCESS', color='$mint'),
    ])),
]


@app('/demo')
async def serve(q: Q):
    q.page['example'] = ui.form_card(box='1 1 6 7', items=[
        ui.table(
            name='issues',
            columns=columns,
            rows=[ui.table_row(name=issue.id, cells=[issue.text, issue.tag]) for issue in issues],
        )
    ])
    await q.page.save()
```

I don't know how we would adapt the API for `FormCard` without breaking changes. I agree that they do look the same though from a user's perspective and it would be better to unify them somehow. Do you have any suggestions?

**Review by mturoci — APPROVED** [review]
Thanks @aalencar!

**Pull request merged**
